pytorch_lightning/trainer/trainer.py

```diff
@@ -122,7 +122,7 @@ def __init__(
         accumulate_grad_batches: Union[int, Dict[int, int]] = 1,
         max_epochs: Optional[int] = None,
         min_epochs: Optional[int] = None,
-        max_steps: Optional[int] = None,
+        max_steps: int = -1,
```

Comment:
this is a breaking change in case, for whatever reason, someone is passing `max_steps=None` explicitly into the trainer constructor. 

we could leave this as Optional[int] = -1
and if someone is passing this as `None`, we log a deprecation warning, and default to using -1 instead.